Content.Shared/StatusEffect/RefCountComponent.cs

```diff
@@ -0,0 +1,18 @@
+namespace Content.Shared.StatusEffect;
+
+/// <summary>
+/// Refcounted components that get only get removed when its sources count goes to 0.
+/// Use this instead of blindly adding/removing components (related: https://youtu.be/k0MUj34y5Kg)
+/// </summary>
+[RegisterComponent, Access(typeof(RefCountSystem))]
+public sealed partial class RefCountComponent : Component
+{
+    /// <summary>
+    /// Each component added and the number of sources it has.
+    /// </summary>
+    /// <remarks>
+    /// Not networked because dear god that would be complicated.
```

Comment:
How so.

Also there's no guards being made for the fact there's no guarantee the client doesn't skip the code anyway if ticks are dropped.

Reply:
if client drops ticks it cant be predicted at all no

Reply:
networked and using component name maybe itll work